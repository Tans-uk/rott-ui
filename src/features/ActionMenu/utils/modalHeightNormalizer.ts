import {display} from '@utils'

export const modalHeightNormalizer = (
  dataLength: number,
  titleExist: boolean,
  subTitleExist: boolean,
  separatorTotalHeight: number,
  maxItem: number,
  itemHeight: number
) => {
  const showItemCount = dataLength > maxItem ? maxItem : dataLength
  const headerHeight = display.px(56)
  const cancelButtonHeight = display.px(56)
  const cancelButtonMarginTop = display.px(16)
  const listBottomPosition = display.px(24)

  return (
    ((showItemCount * itemHeight +
      (titleExist || subTitleExist ? headerHeight : 0) +
      cancelButtonHeight +
      cancelButtonMarginTop +
      listBottomPosition +
      separatorTotalHeight) *
      100) /
    display.setHeight(100)
  )
}
