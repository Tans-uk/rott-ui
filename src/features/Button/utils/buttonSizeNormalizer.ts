import {useDisplay} from '../../../hooks'
import type {Size} from '../../../models'

export const buttonSizeNormalizer = (buttonSize: Size | {width: string; height: string}) => {
  const {px, fontPixel} = useDisplay()

  switch (buttonSize) {
    case 'xs':
      return {
        width: px(85.5),
        height: px(36),
        fontSize: fontPixel(12),
        icon: px(18),
      }
    case 'sm':
      return {
        width: px(114),
        height: px(40),
        fontSize: fontPixel(14),
        icon: px(20),
      }
    case 'md':
      return {
        width: px(171),
        height: px(48),
        fontSize: fontPixel(14),
        icon: px(24),
      }
    case 'lg':
      return {
        width: px(228),
        height: px(56),
        fontSize: fontPixel(16),
        icon: px(24),
      }
    case 'xl':
    case 'xxl':
    case 'full':
    default:
      return {
        width: px(342),
        height: px(56),
        fontSize: fontPixel(16),
        icon: px(24),
      }
  }
}
