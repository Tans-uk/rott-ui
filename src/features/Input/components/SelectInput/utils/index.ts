export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
export * from './sortListBySearchPriority'
