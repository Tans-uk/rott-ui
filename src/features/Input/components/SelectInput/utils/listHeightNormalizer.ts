import {ITEM_HEIGHT, LIST_MAX_ITEM_COUNT, LIST_MIN_ITEM_COUNT, SEPARATOR_HEIGHT} from '../constants'

/**
 * Liste Yuksekligini Tanimlanan Min-Max Item Degerlerine Gore Hesaplar.
 * @param itemCount eleman sayisi
 * @returns liste yuksekligi
 */
export const listHeightNormalizer = (
  itemCount: number,
  showDescription: boolean,
  itemHeight: number = 56
) => {
  if (itemCount < LIST_MIN_ITEM_COUNT) {
    return (
      ITEM_HEIGHT(showDescription, itemHeight) * LIST_MIN_ITEM_COUNT +
      SEPARATOR_HEIGHT * (LIST_MIN_ITEM_COUNT + 1)
    )
  } else if (itemCount >= LIST_MAX_ITEM_COUNT) {
    return (
      ITEM_HEIGHT(showDescription, itemHeight) * LIST_MAX_ITEM_COUNT +
      SEPARATOR_HEIGHT * (LIST_MAX_ITEM_COUNT + 1)
    )
  } else
    return ITEM_HEIGHT(showDescription, itemHeight) * itemCount + SEPARATOR_HEIGHT * (itemCount + 1)
}
