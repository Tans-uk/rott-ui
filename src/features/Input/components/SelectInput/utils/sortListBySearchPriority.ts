import type {SelectProps} from '../models'

import {searchTextNormalizer, searchTextWithTRNormalizer} from '@utils'

export const sortListBySearchPriority = (list: SelectProps[], searchText: string) => {
  const normalizedSearchText = searchTextNormalizer(searchText)
  const normalizedSearchTextWithTR = searchTextWithTRNormalizer(searchText)

  return list.sort((a, b) => {
    const normalizedLabelA = searchTextWithTRNormalizer(a?.label || '')
    const normalizedLabelB = searchTextWithTRNormalizer(b?.label || '')

    // getMatchPriority fonksiyonu, bir metnin arama metniyle olan ilişkisini belirler ve öncelik puanı döndürür.
    const getMatchPriority = (text: string) => {
      const textSearchWithoutTR = searchTextNormalizer(text)

      // Metni normalize ederek tekrar eden işlemleri önler
      const isExactMatchWithTR = text === normalizedSearchTextWithTR
      const isExactMatch = textSearchWithoutTR === normalizedSearchText
      const isStartsWithTR = text.startsWith(normalizedSearchTextWithTR)
      const isStartsWith = textSearchWithoutTR.startsWith(normalizedSearchText)
      const isIncludes = text.includes(normalizedSearchText)

      // Koşullara göre öncelik puanı döndürür
      // Başlangıç eşleşmesi için en yüksek öncelik
      if (isStartsWithTR) return 5
      // Tam eşleşme
      else if (isExactMatch) return 4
      // Tam eşleşme (TR)
      else if (isExactMatchWithTR) return 3
      // Başlangıç eşleşmesi
      else if (isStartsWith) return 2
      // İçerik eşleşmesi
      else if (isIncludes) return 1
      // Eşleşme yoksa öncelik sıfır
      else return 0
    }

    const priorityA = getMatchPriority(normalizedLabelA)
    const priorityB = getMatchPriority(normalizedLabelB)

    // If priorities differ, sort by priority
    if (priorityA !== priorityB) return priorityB - priorityA

    // If priorities are the same, sort alphabetically
    return normalizedLabelA.localeCompare(normalizedLabelB)
  })
}
