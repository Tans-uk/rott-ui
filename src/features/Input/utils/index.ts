export * from './formatByDateMode'
export * from './inputNormalizer'
