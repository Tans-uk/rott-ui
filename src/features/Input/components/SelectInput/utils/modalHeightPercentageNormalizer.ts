import {HEADER_HEIGHT, LIST_BOTTOM_MARGIN} from '../constants'
import {listHeightNormalizer} from './listHeightNormalizer'

import {display} from '@utils'

/**
 * Modal Yuksekligini Eleman Sayisina Gore Yuzde (%) Olarak Hesaplar
 * @param itemCount eleman sayisi
 * @returns Modal Yukseklik Yuzdesi
 */
export const modalHeightPercentageNormalizer = (
  itemCount: number,
  searchable: boolean,
  showDescription: boolean,
  itemHeight: number = 56
) => {
  const listHeight = listHeightNormalizer(itemCount, showDescription, itemHeight)

  const modalHeightPX = display.normalize(
    listHeight + HEADER_HEIGHT(searchable) + LIST_BOTTOM_MARGIN,
    'height'
  )

  return (modalHeightPX * 100) / display.percentage(100, 'height') + 1
}
