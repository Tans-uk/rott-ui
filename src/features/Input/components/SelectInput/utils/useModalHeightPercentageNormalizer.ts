import {useDisplay} from '../../../../../hooks'
import {HEADER_HEIGHT, LIST_BOTTOM_MARGIN} from '../constants'
import {listHeightNormalizer} from './listHeightNormalizer'

/**
 * Modal Yuksekligini Eleman Sayisina Gore Yuzde (%) Olarak Hesaplar
 * @param itemCount eleman sayisi
 * @returns Modal Yukseklik Yuzdesi
 */
export const useModalHeightPercentageNormalizer = (
  itemCount: number,
  searchable: boolean,
  showDescription: boolean,
  itemHeight: number = 56
) => {
  const {normalize, percentage} = useDisplay()
  const listHeight = listHeightNormalizer(itemCount, showDescription, itemHeight)

  const modalHeightPX = normalize(
    listHeight + HEADER_HEIGHT(searchable) + LIST_BOTTOM_MARGIN,
    'height'
  )

  return (modalHeightPX * 100) / percentage(100, 'height') + 1
}
