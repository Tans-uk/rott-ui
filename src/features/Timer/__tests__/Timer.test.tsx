import {TimerComponent} from '@features'
import {render} from 'src/__tests__/utils/testUtils'

describe('Timer -> Custom Component', () => {
  const timerInitialTime = 180

  beforeAll(() => {
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  it('timer ilk render anında snapshot ile eşleşmeli', () => {
    const renderedTimer = render(<TimerComponent time={timerInitialTime} />)

    expect(renderedTimer).toMatchSnapshot()
  })

  it('verilen time doğru renderlanmalı', async () => {
    const {getByTestId} = render(<TimerComponent time={timerInitialTime} />)

    const label = getByTestId('timerTestId')

    expect(label).toHaveProp('children', '3:00')
  })
})
