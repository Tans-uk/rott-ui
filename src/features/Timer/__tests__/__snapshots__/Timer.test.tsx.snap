// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer -> Custom Component timer ilk render anında snapshot ile eşleşmeli 1`] = `
<View
  flex={1}
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <KeyboardProvider
    enabled={true}
  >
    <View
      flex={1}
      pointerEvents="box-none"
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "flex": undefined,
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgba(17, 17, 17, 1)",
                "fontFamily": "Markpro-Bold",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "auto",
              }
            }
            testID="timerTestId"
          >
            3:00
          </Text>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          [
            {
              "alignSelf": "center",
              "elevation": 999999,
              "flex": 0,
              "maxWidth": "100%",
              "position": "absolute",
              "zIndex": 999999,
            },
            {
              "flexDirection": "column-reverse",
              "justifyContent": "flex-start",
              "top": 30,
              "width": 750,
            },
          ]
        }
      >
        <View
          style={
            {
              "bottom": 0,
            }
          }
        >
          <RCTSafeAreaView />
        </View>
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          [
            {
              "alignSelf": "center",
              "elevation": 999999,
              "flex": 0,
              "maxWidth": "100%",
              "position": "absolute",
              "zIndex": 999999,
            },
            {
              "bottom": 40,
              "flexDirection": "column",
              "justifyContent": "flex-end",
              "width": 750,
            },
          ]
        }
      >
        <View
          style={
            {
              "bottom": 0,
            }
          }
        >
          <RCTSafeAreaView />
        </View>
      </View>
    </View>
  </KeyboardProvider>
</View>
`;
